src/Common/src/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -136,6 +136,24 @@ private MethodIL EmitIL()
             return new  PInvokeILStubMethodIL((ILStubMethodIL)emitter.Link(targetMethod), IsStubRequired());
         }
 
+        public static MethodIL EmitExceptionBody(string message, MethodDesc method)
```

Comment:
Good Point. I'll add a TODO to use ThrowHelpers infrastructure to generate the error messages. Also another TODO for logging warnings.